sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -2419,6 +2435,26 @@ private <T extends Resource> Flux<FeedResponse<T>> createReadManyQuery(
         return executionContext.flatMap(IDocumentQueryExecutionContext<T>::executeAsync);
     }
 
+    private <T> Flux<FeedResponse<Document>> createPointReadOperations(
+        Map<PartitionKeyRange, CosmosItemIdentity> singleItemPartitionRequestMap,
+        String resourceLink,
+        CosmosQueryRequestOptions queryRequestOptions,
+        Class<T> klass
+    ) {
+        return Flux.fromIterable(singleItemPartitionRequestMap.values())
+            .flatMap(item -> {
+                CosmosQueryRequestOptions clonedQueryRequestOptions = ModelBridgeInternal.createQueryRequestOptions(queryRequestOptions);
+                clonedQueryRequestOptions.setPartitionKey(item.getPartitionKey());
+                return this.readDocument(getDocumentLink(item.getId(), resourceLink), ModelBridgeInternal.toRequestOptions(clonedQueryRequestOptions));
```

Comment:
Since we are doing a point read here, I think it would be better if we can convert the `CosmosQueryRequestOptions` to `CosmosItemRequestOptions` and pass those to point reads. That way, we can add / modify any point read related options if we need to. 

`CosmosItemRequestOptions cosmosItemRequestOptions = new CosmosItemRequestOptions();`